File: Week 2/Assignment 1/grader.py
class Grader:
  #constructor, sets variable based on input
  def __init__(self): 
    self.student = input("Enter Student Name: ")
    self.assignment = input("Enter Assignment: ")
    self.grade = self.getGrade()

  #function to get the grade, validate, and convert to letter.
  def getGrade(self):
    #loop to get correct input
    while True:
      try: 
        grade = float(input("Enter Score: "))
      except ValueError:
        #not a number, loop again
        print("Please only enter a number score.")
        continue
      if grade < 0 or grade > 100:
        #number out of bounds, loop again
        print("Please only enter a score between 0 and 100")
        continue
      else:
        #input correct, exit loop
        break
    
    #letter grade var
    letterGrade = "A"

    #sets lettergrade based on score.
    if grade < 60:
      letterGrade = "F"
    elif grade < 70:
      letterGrade = "D"
    elif grade < 80:
      letterGrade = "C"
    elif grade < 90:
      letterGrade = "B"

    return letterGrade

  #function to output result.
  def outputGrade(self):
      print("\n- Student Name: " + self.student + "\n- Assignment: " + self.assignment + "\n- Grade: " + self.grade) 

#create object
grader1 = Grader()
#output 
grader1.outputGrade()

File: Week 2/Assignment 1/grader.rb
class Grader
  # constructor, sets values on run
  def initialize()
    puts "Enter Student Name: "
    @student = gets
    puts "Enter Assignment: "
    @assignment = gets
    @grade = getGrade()
  end

  # function to output data
  def outputGrade
    return "\nStudent Name: "+ @student +"Assignment: "+ @assignment +"Grade: "+@grade
  end

  #function to get the score, validate and convert to letter. 
  def getGrade
    puts "Enter Score: "
    #gets score as float, if not float, sets to false.
    score = Float(gets) rescue false

    #loop for valid input, if not false, or between 0 and 100
    while !score or score > 100 or score < 0 do
      puts "Please only enter a number between 0 and 100."
      puts "Enter Score: "
      score = Float(gets) rescue false
    end

    # convert to letter
    letterGrade = "A"
    if score < 60
      letterGrade = "F"
    elsif score < 70
      letterGrade = "D"
    elsif score < 80
      letterGrade = "C"
    elsif score < 90
      letterGrade = "B"
    end
    return letterGrade
  end
end

#instantiate, call output function
g = Grader.new
print(g.outputGrade())






